Editorial review: Document CSS multicol column wrapping (#42934)
mdn/content: files/en-us/web/css/reference/properties/columns/index.md

columns: unset;
```

> The `columns` property may be specified as a [`<column-count>`](/en-US/docs/Web/CSS/Reference/Properties/column-count) and/or a [`<column-width>`](/en-US/docs/Web/CSS/Reference/Properties/column-width) value, optionally followed by a forward slash and a [`<column-height>`](/en-US/docs/Web/CSS/Reference/Properties/column-height) value. If both `<column-count>` and `<column-width>` values are specified, they can be specified in any order.

Reviewer: Alternatively:

```suggestion
The `columns` property may be specified as a [`<column-count>`](/en-US/docs/Web/CSS/Reference/Properties/column-count) and/or a [`<column-width>`](/en-US/docs/Web/CSS/Reference/Properties/column-width) value.  As the count is an integer and the width is a length, these components can be declared in any order.  If included, the optional [`<column-height>`](/en-US/docs/Web/CSS/Reference/Properties/column-height) value must be preceded by a forward slash (`/`).
```

> The `columns` property may be specified as a [`<column-count>`](/en-US/docs/Web/CSS/Reference/Properties/column-count) and/or a [`<column-width>`](/en-US/docs/Web/CSS/Reference/Properties/column-width) value, optionally followed by a forward slash and a [`<column-height>`](/en-US/docs/Web/CSS/Reference/Properties/column-height) value. If both `<column-count>` and `<column-width>` values are specified, they can be specified in any order.

Author: I played around a bit, and ended up with 

> The `columns` property value may be specified as a [`<column-count>`](/en-US/docs/Web/CSS/Reference/Properties/column-count) and/or a [`<column-width>`](/en-US/docs/Web/CSS/Reference/Properties/column-width) value, in any order, optionally followed by a [`<column-height>`](/en-US/docs/Web/CSS/Reference/Properties/column-height) value preceded by a forward slash.

### Values

- `<'column-width'>`
  - : The ideal column width, defined as a {{cssxref("&lt;length&gt;")}} or the keyword `auto`. The actual width may be wider or narrower to fit the available space. See {{cssxref("column-width")}}.
- `<'column-count'>`
  - : The ideal number of columns into which the element's content should be flowed, defined as an {{cssxref("&lt;integer&gt;")}} or the keyword `auto`. If neither this value nor the column's width are `auto`, it indicates the maximum allowable number of columns. See {{cssxref("column-count")}}.
- `<'column-height'>`
  - : The height of the columns defined as a {{cssxref("&lt;length&gt;")}} or the keyword `auto`. See {{cssxref("column-height")}}.

> [!NOTE]

> > In browsers that support the `column-height` property, the `columns` property also resets the {{cssxref("column-wrap")}} property to its initial value, `auto`.

Reviewer: ```suggestion
> Setting a `<column-height>` resets the {{cssxref("column-wrap")}} property to its initial value, `auto`. When set to a `<length>`, the {{cssxref("column-wrap")}} behaves as `wrap`. When set to `auto`, it behaves as `nowrap`.
```
because of the slash requirement the declaration is ignored when not supported, so the above should be ok and future proofed.

Author: Played around with it, and ended up with:

> Setting a `<column-height>` value resets the {{cssxref("column-wrap")}} property to its initial value, `auto`. When `<column-height>` is set to a `<length>`, the {{cssxref("column-wrap")}} `auto` value behaves as `wrap`. When `<column-height>` is set to `auto`, it behaves as `nowrap`.

### Setting three equal columns

This example shows how to split a text container into three equal columns.

#### HTML

We include a basic {{htmlelement("p")}} element containing text content.

```html live-sample___three_equal_columns
<p class="content-box">
  This is a bunch of text split into three columns using the CSS
  <code>columns</code>
  property. The text is equally distributed over the columns.
</p>
```

#### CSS

We set a `columns` property value on the paragraph that includes a `<column-count>` value of `3` and a `<column-width>` value of `auto`.

```css hidden live-sample___three_equal_columns
body {
  width: 60%;
  margin: 0 auto;
}
```

```css live-sample___three_equal_columns
.content-box {
  columns: 3 auto;
}
```

#### Result

The example renders like this:

> 

Reviewer: ```suggestion
```
limit these intros to guides where there are no in-example headings. Omit after "result" in reference pages

  <code>column-wrap: auto</code> resolves to <code>wrap</code>, which allows the
  columns to wrap onto multiple rows. The text is equally distributed over the
  columns, and placed into multiple rows.
</p>
```

#### CSS

We set a `columns` property value on the paragraph that includes a `<column-count>` value of `3`, a `<column-width>` value of `auto`, and a `<column-height>` value of `5em`. Because the `<column-height>` is set to a `<length>`, the paragraph's {{cssxref("column-wrap")}} value computes to `wrap`, which allows the columns to wrap onto multiple rows.

```css hidden live-sample___fixed-height
body {
  width: 60%;
  margin: 0 auto;
}
```

```css live-sample___fixed-height
.content-box {
  columns: 3 auto / 5em;
}
```

```css hidden live-sample___fixed-height
@supports not (columns: 3 auto / 5em) {
  body::before {
    content: "Your browser does not support the 'column-height' property.";
    background-color: wheat;
    position: fixed;
    inset: 40% 0;
    height: fit-content;
    text-align: center;
    padding: 1rem 0;
  }
}
```

#### Result

The example renders like this:

{{EmbedLiveSample('fixed-height', 'auto', 320)}}

> Note how the text is split up into three columns. Each column is `5em` height. After every third column, the columns wrap onto a new line in the block direction.

Reviewer: ```suggestion
Note how the text is split up into three columns. Each column is `5em` tall. When the three columns overflow the defined height, the columns wrap onto a new line in the block direction.
```

{{EmbedLiveSample('Setting_three_equal_columns', 'auto', 120)}}

### Setting three wrapping, fixed height columns

#### HTML

```html

> <p class="content-box">

Author: added

  This is a bunch of text split into three columns using the CSS
  <code>columns</code> property. This includes a <code>column-count</code> value
  of <code>3</code>, a <code>column-width</code> value of <code>auto</code>, and
  a <code>column-height</code> value of <code>5em</code>. The
  <code>column-wrap</code> value is set to its initial value, <code>auto</code>;
  when a <code>column-height</code> value is included,
  <code>column-wrap: auto</code> resolves to <code>wrap</code>, which allows the
  columns to wrap onto multiple rows. The text is equally distributed over the
  columns, and placed into multiple rows.
</p>
```

#### CSS

> 

Reviewer: explanation
